engines/python/setup/djl_python/deepspeed.py

```diff
@@ -259,28 +377,25 @@ def create_model_pipeline(self):
             lora_model = PeftModel.from_pretrained(base_model,
                                                    self.model_id_or_path)
             model = lora_model.merge_and_unload()
+            self.tokenizer = AutoTokenizer.from_pretrained(self.peft_config.base_model_name_or_path)
             self.logger.info(
                 f"Peft Model merged into base model for deepspeed compatibility"
             )
         else:
-            model = TASK_TO_MODEL[self.task].from_pretrained(
+            model, self.tokenizer, state_dict_mmap = self.load_model(
                 self.model_id_or_path,
+                'pretrained',
+                self.ds_config['replace_with_kernel_inject'],
                 low_cpu_mem_usage=self.low_cpu_mem_usage,
                 trust_remote_code=self.trust_remote_code,
-                **kwargs)
+                **kwargs
+            )
         if self.data_type:
             self.ds_config["dtype"] = self.data_type
         else:
             self.ds_config["dtype"] = model.dtype
-        if self.model_config.model_type in OPTIMIZED_MODEL_TYPES:
-            self.ds_config["replace_with_kernel_inject"] = True
-        self.model = deepspeed.init_inference(model, config=self.ds_config)
-        if self.peft_config:
-            self.tokenizer = AutoTokenizer.from_pretrained(
-                self.peft_config.base_model_name_or_path)
-        else:
-            self.tokenizer = AutoTokenizer.from_pretrained(
-                self.model_id_or_path)
+        self.model = deepspeed.init_inference(model, state_dict_mmap, config=self.ds_config)
```

Comment:
Totally getting rid of peft Lora?

Reply:
`elif self.peft_config is not None:`(ln 368 - ln 384) block should work as before.